sympy/assumptions/ask.py

```diff
@@ -9,66 +9,1114 @@
+        >>> ask(Q.nonzero(0))
+        False
+        >>> ask(Q.nonzero(I)) # doctest: +SKIP
+        False
+        >>> ask(~Q.zero(I))
+        True
+
+        """
+        return Predicate('nonzero')
+
+    @memoize_property(predicate_storage)
+    def nonpositive(self):
+        """
+        Nonpositive real number predicate.
+
+        ``ask(Q.nonpositive(x))`` is true iff ``x`` belongs to the set of
+        negative numbers including zero.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, I
+        >>> ask(Q.nonpositive(-1))
+        True
+        >>> ask(Q.nonpositive(0))
+        True
+        >>> ask(Q.nonpositive(1))
+        False
+        >>> ask(Q.nonpositive(I))
+        False
+        >>> ask(Q.nonpositive(-I))
+        False
+
+        """
+        return Predicate('nonpositive')
+
+    @memoize_property(predicate_storage)
+    def nonnegative(self):
+        """
+        Nonnegative real number predicate.
+
+        ``ask(Q.nonnegative(x))`` is true iff ``x`` belongs to the set of
+        positive numbers including zero.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, I
+        >>> ask(Q.nonnegative(1))
+        True
+        >>> ask(Q.nonnegative(0))
+        True
+        >>> ask(Q.nonnegative(-1))
+        False
+        >>> ask(Q.nonnegative(I))
+        False
+        >>> ask(Q.nonnegative(-I))
+        False
+
+        """
+        return Predicate('nonnegative')
+
+    @memoize_property(predicate_storage)
+    def even(self):
+        """
+        Even number predicate.
+
+        ``ask(Q.even(x))`` is true iff ``x`` belongs to the set of even numbers.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, pi
+        >>> ask(Q.even(0))
+        True
+        >>> ask(Q.even(2))
+        True
+        >>> ask(Q.even(3))
+        False
+        >>> ask(Q.even(pi))
+        False
+
+        """
+        return Predicate('even')
+
+    @memoize_property(predicate_storage)
+    def odd(self):
+        """
+        Odd number predicate.
+
+        ``ask(Q.odd(x))`` is true iff ``x`` belongs to the set of odd numbers.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, pi
+        >>> ask(Q.odd(0))
+        False
+        >>> ask(Q.odd(2))
+        False
+        >>> ask(Q.odd(3))
+        True
+        >>> ask(Q.odd(pi))
+        False
+
+        """
+        return Predicate('odd')
+
+    @memoize_property(predicate_storage)
+    def prime(self):
+        """
+        Prime number predicate.
+
+        ``ask(Q.prime(x))`` is true iff ``x`` is a natural number greater
+        than 1 that has no positive divisors other than ``1`` and the
+        number itself.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask
+        >>> ask(Q.prime(0))
+        False
+        >>> ask(Q.prime(1))
+        False
+        >>> ask(Q.prime(2))
+        True
+        >>> ask(Q.prime(20))
+        False
```

Comment:
Maybe show `ask(Q.prime(-3))` for clarity (since sometimes -3 is defined to be prime).